pkg/inject/inject.go

```diff
@@ -0,0 +1,173 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package inject
+
+import (
+	"fmt"
+	"reflect"
+	"strings"
+)
+
+type aggregateError []error
+
+func (e aggregateError) Error() string {
+	var errStrs []string
+	for _, err := range e {
+		errStrs = append(errStrs, err.Error())
+	}
+	return strings.Join(errStrs, ",")
+}
+
+var errorType = reflect.TypeOf((*error)(nil)).Elem()
+
+// provideCtx provides the given context with or without the
+// key "Dependencies" (to allow avoiding conflict with the normal
+// context object in injection methods)
+func provideCtx(ctx Context) Context {
+	ctxVal := reflect.ValueOf(ctx)
+	return &depContext{
+		key:         "Dependencies",
+		value:       ctxVal,
+		typ:         ctxVal.Type(),
+		keyOptional: true,
+		next:        ctx,
+	}
+}
+
+// Into reads the given object for implementations
+// of injectable pseudo-interfaces and struct tags,
+// and injects dependencies based on those.
+// As a special case, the given context will also "Provide" itself
+// with or without key "Dependencies".
+func Into(ctx Context, target interface{}) (bool, error) {
```

Comment:
I feel the order of `ctx` and `target` needs to be reversed.

Reply:
yeah.. but everything else takes ctx first.